### PR TITLE
Remove duplicate packages

### PR DESCRIPTION
**Pull request by jhadvig**
Rules out the possibility that install GA and upgrade to latest will point to the same version.
Additional TC's added.
@stevekuznetsov PTAL

### CODE REVIEW

File: sjb/hack/determine_install_upgrade_version.py [hunk 1]
```diff
@@ -3,6 +3,16 @@
 import os.path
 import rpmUtils.miscutils as rpmutils
 
+# Remove duplicate packages from different repositories
+def remove_duplicate_pkgs(available_pkgs):
```
Comment (stevekuznetsov) [inline sjb/hack/determine_install_upgrade_version.py]: Will it work just to use a `set()` here?

Comment (jhadvig) [inline sjb/hack/determine_install_upgrade_version.py]: unfortunately no. The thing is that the pkgs that point to the same verions are pointing to a different repo, that why `set()` wont treat them as dups.

Comment (stevekuznetsov) [inline sjb/hack/determine_install_upgrade_version.py]: ack

File: sjb/hack/determine_install_upgrade_version.py [hunk 2]
```diff
@@ -80,8 +90,9 @@ def determine_search_version(pkg_name, pkg_version):
 	yb.conf.disable_excludes = ["all"]
```
Comment (stevekuznetsov) [inline sjb/hack/determine_install_upgrade_version.py]: Unrelated to this PR but we should check to see if we can remove this -- was this due to some excluder bug with them being on all the time or are we always running this script when they are on and we need to keep it?

**Comment by stevekuznetsov** [comment]
LGTM

File: sjb/hack/determine_install_upgrade_version.py [hunk 3]
```diff
@@ -80,8 +90,9 @@ def determine_search_version(pkg_name, pkg_version):
...
 	sys.stdout = old_stdout
 	available_pkgs = rpmutils.unique(yb.doPackageLists('available', patterns=[pkg_name], showdups=True).available)
-	available_pkgs.sort(lambda x, y: rpmutils.compareEVR((x.epoch, x.version, x.release), (y.epoch, y.version, y.release)))
 	search_version = determine_search_version(pkg_name, pkg_version)
+	available_pkgs = remove_duplicite_pkgs(available_pkgs)
```
Comment (smarterclayton) [inline sjb/hack/determine_install_upgrade_version.py]: typo, `duplicate`

**Comment by smarterclayton** [comment]
One comment about typo

**Comment by jhadvig** [comment]
@smarterclayton comment addressed

**Comment by smarterclayton** [comment]
LGTM as well

**Pull request merged**